cocotb/triggers.py

```diff
@@ -735,3 +735,44 @@ def _wait(self):
         for _ in range(self.num_cycles):
             yield trigger
         raise ReturnValue(self)
+
+
+@decorators.coroutine
+def with_timeout(triggers, timeout_time, timeout_unit=None):
+    """
+    Waits on triggers, throws an exception if it waits longer than the given time
+
+    Can be used with any expression that would otherwise appear to the right of a
+    `yield` (or `await` in Python 3) statement.
+
+    Usage:
+
+    .. code-block:: python
+
+        yield with_timeout(coro, 100, 'ns')
+        yield with_timeout([coro, event.wait(), ...], 100, 'ns')
+
+    Args:
+        triggers (cocotb_yield_expr):
+            Any expression that could be right of a yield statement in cocotb
```

Comment:
It's a shame we don't really have a concise way to describe _"a thing accepted by `_trigger_from_any`"_. No action needed here.